tools/pipelines/templates/build-npm-package.yml

```
@@ -605,35 +603,49 @@ stages:
         publishNonScopedPackages: ${{ variables.publishNonScopedPackages }}
 
   # Capture per-pipeline stage results
-  - stage: runAfterAll
-    displayName: Retrieve and upload client stage stats to Kusto
+  - stage: upload_run_telemetry
+    displayName: Upload pipeline run telemetry to Kusto
     condition: and(succeededOrFailed(), eq(variables['Build.Reason'], 'IndividualCI'), eq(${{parameters.telemetry}}, true))
-    dependsOn: 
+    dependsOn:
       - build
 
     jobs:
-      - template: include-telemetry-setup.yml
-        parameters:
-            testWorkspace: ${{ variables.pipelineTelemetryWorkdir }}
-            poolBuild: Small
-            customSteps:
+      - job: upload_run_telemetry
+        displayName: Upload pipeline run telemetry to Kusto
+        pool: Small
+        variables:
+        - name: pipelineTelemetryWorkdir
+          value: $(Pipeline.Workspace)/pipelineTelemetryWorkdir
 
-            - task: Bash@3
-              displayName: Retrieve buildId results
-              inputs:
-                targetType: 'inline'
-                workingDirectory: $(toolAbsolutePath)
-                script: |
-                  echo "creating output folder"
-                  mkdir -p ${{ variables.pipelineTelemetryWorkdir }}/timingOutput
-                  echo "Executing curl command ..."
-                  echo  'curl -u ":<REDACTED>" "https://dev.azure.com/fluidframework/internal/_apis/build/builds/$BUILD_ID/timeline"'
-                  curl -u ":$ADO_API_TOKEN" "https://dev.azure.com/fluidframework/internal/_apis/build/builds/$BUILD_ID/timeline\?api-version\=6.0-preview.1" > ${{ variables.pipelineTelemetryWorkdir }}/timingOutput/output.json
-                  pwd;
-                  ls -laR ${{ variables.pipelineTelemetryWorkdir }}/timingOutput/output.json;
-                  cat ${{ variables.pipelineTelemetryWorkdir }}/timingOutput/output.json;
-                  node --require @ff-internal/aria-logger bin/run --handlerModule $(Build.SourcesDirectory)/tools/telemetry-generator/dist/handlers/stageTimingRetriever.js --dir '${{ variables.pipelineTelemetryWorkdir }}/timingOutput/';
-              env:
-                BUILD_ID: $(Build.BuildId)
-                ADO_API_TOKEN: $(System.AccessToken)
-                PIPELINE: 'BuildClient'
+        steps:
+        - template: include-telemetry-setup.yml
+          parameters:
+            devFeedUrl: $(ado-feeds-dev)
+            officeFeedUrl: $(ado-feeds-office)
+
+        - task: Bash@3
+          displayName: Create working directory
+          inputs:
+            targetType: 'inline'
+            script: |
+              mkdir $(pipelineTelemetryWorkdir)
+
+        - task: Bash@3
+          displayName: Retrieve buildId results
+          inputs:
+            targetType: 'inline'
+            workingDirectory: $(toolAbsolutePath)
+            script: |
+              echo "creating output folder"
```

Nit (to be more in line with our other inner step logging):
```suggestion
              echo "Creating output folder ..."
```